pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -820,6 +823,44 @@ func (og *operationGenerator) GenerateVerifyControllerAttachedVolumeFunc(
 	}, nil
 }
 
+func (og *operationGenerator) GenerateVerifyControllerDetachedVolumeFunc(
+	volumeToDetach AttachedVolume,
+	mounter mount.Interface,
+	actualStateOfWorld ActualStateOfWorldAttacherUpdater) (func() error, error) {
+	return func() error {
+		if !volumeToDetach.PluginIsAttachable {
+			// If the volume does not implement the attacher interface, it is
+			// assumed to be detached and the actual state of the world is
+			// updated accordingly.
+			actualStateOfWorld.MarkVolumeAsDetached(volumeToDetach.VolumeName, volumeToDetach.NodeName)
+			return nil
+		}
+
+		attached, err := mounter.PathIsDevice(volumeToDetach.DevicePath)
```

Comment:
Just a note that PathIsDevice check might hang for nfs mount when server is down. Since this check is in a gorountine, it should not affect other operations. NFS hanging problem is still under investigation